hcn/hcn.go

```diff
@@ -134,6 +134,15 @@ func V2SchemaVersion() SchemaVersion {
 	}
 }
 
+// RemoteSubnetSupported returns an error if the HCN version does not support Remote Subnet policies.
+func RemoteSubnetSupported() error {
```

Comment:
Why cant the caller do this?

Reply:
The caller could, but if there are multiple checks are required it can save space to have a one line function call.

Reply:
The caller can, but I wanted to reuse the platformnotsupported error.

Reply:
I am ok with leaving it I just thought it seemed easier from a callers perspective to use:

```go
if f := hcn.GetSupportedFeatures(); f.RemoteSubnet {
  // do something
}
```

Doesn't really bother me either way though your call.

Reply:
@erfrimod/@ksubrmnn - Your call here. Let me know and we can merge or update.

Reply:
Your way does look nice and clean... Let's check this in for now. Maybe see how CNI implementations are checking supported features and do a polish pass later.